### PR TITLE
feat:bns service

### PR DESCRIPTION
**Pull request by johnsonwyj**

### CODE REVIEW

File: internal/executor/contracts/service_registry.go [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+		SonName:     sonName,
+		Onwer:       owner,
+		Resolver:    resolver,
+	}
+	subDomainProposalDataBytes, err := json.Marshal(subDomainProposalData)
+	if err != nil {
+		return boltvm.Error(boltvm.BnsErrCode, fmt.Sprintf(string(boltvm.BnsErrMsg), "marshal err"))
+	}
+	event := governance.EventRegister
+	proposalRes := sr.CrossInvoke(constant.GovernanceContractAddr.Address().String(), "SubmitProposal",
```
Comment (Karenlrx) [inline internal/executor/contracts/service_registry.go:185]: why proposal needn't invoke function of 
sr.CrossInvoke(constant.GovernanceContractAddr.Address().String(), "ZeroPermission", pb.String(string(proposalRes.Result))) ?

File: internal/executor/contracts/service_registry.go [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
+type SubDomainProposalData struct {
+	ParentName  string `json:"parent_name"`
+	ParentOwner string `json:"parent_owner"`
+	SonName     string `json:"son_name"`
+	Onwer       string `json:"owner"`
+	Resolver    string `json:"resolver"`
+}
+
+//core func
+//注册1级域名
```
Comment (Karenlrx) [inline internal/executor/contracts/service_registry.go:62]: English code comments is better

File: internal/executor/contracts/service_registry_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
+package contracts
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"github.com/golang/mock/gomock"
```
Comment (Karenlrx) [inline internal/executor/contracts/service_registry_test.go:6]: sort imports

File: internal/executor/contracts/service_registry.go [hunk 4]
```diff
@@ -0,0 +1,594 @@
+package contracts
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"github.com/meshplus/bitxhub-core/boltvm"
```
Comment (Karenlrx) [inline internal/executor/contracts/service_registry.go:7]: sort imports

File: internal/executor/contracts/service_resolver.go [hunk 5]
```diff
@@ -0,0 +1,228 @@
+package contracts
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/meshplus/bitxhub-core/boltvm"
```
Comment (Karenlrx) [inline internal/executor/contracts/service_resolver.go:6]: above all

**Review by Karenlrx — COMMENTED** [review]
rebase commit, merge some wrong commit to one correct commit

**Review by Karenlrx — APPROVED** [review]

**Review by pigxiaolei — APPROVED** [review]

**Pull request merged**